### PR TITLE
Use deepPath for event path trimming.

### PR DESCRIPTION
**Pull request by TakayoshiKochi**
What this test (test-003.html) does is now specced as
"Event Path Trimming", and when retargeted "target" and
"relatedTarget" become same, not only trusted events are
not dispatched, Event.deepPath is also trimmed for these
nodes.

Now the test uses .deepPath rather than relying on
generating "trusted" events and checking if event handler
is not invoked.

### CODE REVIEW

**Comment by TakayoshiKochi** [comment]
@hayatoito PTAL

**Comment by TakayoshiKochi** [comment]
Ping?

File: shadow-dom/events/event-dispatch/test-003.html [hunk 1]
```diff
@@ -29,31 +29,53 @@
...
 
+    var input2 = d.createElement('input');
+    input2.setAttribute('id', 'input2');
+    s.appendChild(input2);
 
-    var evt = document.createEvent("MouseEvents");
-    evt.initMouseEvent("mouseover", true, false, window,
-      0, 10, 10, 10, 10, false, false, false, false, 0, div1);
+    // Add an event listener in its shadow root.
+    s.addEventListener('focusin', A_05_05_03_T01.step_func(function(event) {
```
Comment (hayatoito) [inline shadow-dom/events/event-dispatch/test-003.html]: Why don't you add an event listener to a input1, or input2, to avoid `if elseif else`?

**Comment by TakayoshiKochi** [comment]
Please take a look at #2130.

**Pull request closed without merging**